go-controller/pkg/ovn/pods.go

```diff
@@ -734,20 +734,20 @@ func (oc *DefaultNetworkController) addLogicalPort(pod *kapi.Pod) (err error) {
 	return nil
 }
 
-func (oc *DefaultNetworkController) updatePodAnnotationWithRetry(origPod *kapi.Pod, podInfo *util.PodAnnotation) error {
+func (oc *DefaultNetworkController) updatePodAnnotationWithRetry(origPod *kapi.Pod, podInfo *util.PodAnnotation, replaceExistingNet bool) error {
 	resultErr := retry.RetryOnConflict(util.OvnConflictBackoff, func() error {
-		// Informer cache should not be mutated, so get a copy of the object
-		pod, err := oc.watchFactory.GetPod(origPod.Namespace, origPod.Name)
+		// We don't retrieve the pod from the pod informer cache because if the pod informer cache is not up-to-date,
+		// we risk replacing or overwriting an existing network specified within an annotation.
+		pod, err := oc.kube.GetPod(origPod.Namespace, origPod.Name)
```

Comment:
@martinkennelly @trozet this is gonna kill PodReadyLatency in scale tests.

Reply:
@dcbw Yeah. Any ideas?  
We maybe moving to using a CRD to define our networks instead of using annotations. I dont understand how that would help though but maybe you've an idea.